### ISSUE TITLE
Plugin Manager reverts status of all plugins to "Not Installed" when bad plugin loaded

### ISSUE THREAD

**Issue by OldNGreyBMR** [zencart/zencart#6884]
**Describe the bug**
Plugins are installed and operating. If a new plugin that contains an error is added, all plugins are shown as "Not Installed"

**Version**
Zen Cart version: 2.1.0
PHP version: 8.1 to 8.4
MySQL version: MariaDb 11.16.2 and MariaDb10.6.17
Plugins that may be related: all
What older/newer versions were also tested? zc1.5.8a 

**To Reproduce**
Steps to reproduce/demonstrate the behavior:
1. In admin console install some plugins then logoff
2. Add new plugin to folder with hierarchy of plugin_name folder, no version number folder, manifest file, installer folde retc
3. Log on to admin , Plugins Manager 
4. See plugins are 'Not installed'
**Expected behavior**
Throw an error and write to log file or do nothing and do not display the new plugin as occurs in zc 1.5.8a

**Screenshots**
No plugins
![image](https://github.com/user-attachments/assets/718d6946-319e-4a98-8d33-70a2f5c31a3f)
3 plugins
![image](https://github.com/user-attachments/assets/5252705c-dfca-4c09-af92-70b1d7989873)
Load bad plugin to folder and plugins reset
![image](https://github.com/user-attachments/assets/c19c68e3-df8f-4ea7-8f9e-e66a0fa7e30f)

**Debug/Error Logs**
No deburg/ error logs
 
**Devices (please complete the following information):**
 - Operating System:  Windows 10, 
 - Browser all browsers: Edge Brave, Chrome, Firefox 

**Additional context**
Other error conditions in the new plugin will cause the same result but this one is the most easily reproduced.

**Comment by drbyte** [comment]
Do the symptoms clear if the bad directory or other offending condition is removed?

**Comment by OldNGreyBMR** [comment]
The settings stay as "Not Installed" for all plugins if the offending
condition is removed.

**Comment by drbyte** [comment]
> Add new plugin to folder with hierarchy of plugin_name folder, no version number folder, manifest file, installer folde retc

I tried following your replication instructions, on v2.1.0, but no problems surface. Functions normally, ignoring the bad directory.

To be fair, I don't really understand your instructions.

I created `/zc_plugins/foo/` and no errors occurred.
I added an empty version number directory, and no problems.
I added an empty manifest file to the directory, no problems.
I added an installer folder to the directory, no problems.
